templates/EC2_Quickstart.yml

```diff
@@ -0,0 +1,156 @@
+    Description: Installs latest version NewRelic Infrastructure agent in EC2 instances once every month.
+    Type: 'AWS::SSM::Association'
+    Properties:
+      AssociationName: CT-NewRelic-Infra-Agent-Installer
+      Name: AWS-ConfigureAWSPackage
+      ScheduleExpression: "rate(30 days)"
+      Targets:
+        - Key: "tag:Supervisor"
+          Values:
+            - "new-relic"
+      Parameters:
+        action: 
+          - !Ref Action
+        installationType: 
+          - !Ref InstallationType
+        name: 
+          - !Ref NewRelicInfrastructureAgent
+        additionalArguments: 
+          - !Ref AdditionalParametersLicenseKey
+
+  NewRelicApiDestination:
+    DependsOn: NewRelicConnection
+    Type: AWS::Events::ApiDestination
+    Properties:
+      Name: CT-NewRelic-Logs-Endpoint
+      HttpMethod: POST
+      InvocationEndpoint: !Ref NewRelicLogsEndpoint
+      ConnectionArn: !GetAtt NewRelicConnection.Arn
+
+  NewRelicConnection:
+    Type: AWS::Events::Connection
+    Properties:
+      Name: CT-NewRelic-Endpoint-Connection
+      AuthorizationType: API_KEY
+      AuthParameters:
+        ApiKeyAuthParameters:
+          ApiKeyName: Api-Key
+          ApiKeyValue: !Ref NewRelicLicenseKey
+        InvocationHttpParameters: 
+          HeaderParameters: 
+            - Key: Api-Key
+              Value: !Ref NewRelicLicenseKey
+        
+  EventBridgeApiDestinationPolicy:
+    DependsOn: InvokeApiDestinationEventBridgeRole
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: CT-NewRelic-EventBridgeApiDestinationPolicy
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - "events:InvokeApiDestination"
+            Resource:
+              - !Sub 'arn:aws:events:${AWS::Region}:${AWS::AccountId}:api-destination/CT-Newrelic-Logs-Endpoint/*'
+      Roles:
+        - !Ref InvokeApiDestinationEventBridgeRole
+        
+  InvokeApiDestinationEventBridgeRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: CT-NewRelic-EventBridge-Invoke-Api-Destination
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: events.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonEventBridgeFullAccess
+    
+  SendLogsToNREndpoint:
+    Type: AWS::Events::Rule
+    Properties:
+      Name: CT-NewRelic-Failed-EC2-Infra-Agent-Installations-Logs-Rule
+      EventPattern:
+        source:
+          - "aws.ssm"
+        detail-type:
+          - "EC2 Command Invocation Status-change Notification"
+        detail:
+          status:
+            - "Failed"
+      Targets:
+        - Arn: !GetAtt NewRelicApiDestination.Arn
+          Id: "CT-Newrelic-Logs-Endpoint"
+          RoleArn: !GetAtt InvokeApiDestinationEventBridgeRole.Arn
+          
+  InstallNewRelicAgentRule:

```

Comment:
Wondering why didnt you create another SSM Association (not cron based) for one time execution. Atleast when created like this from UI, a new instance is bootstrapped once with the package.

Reply:
@anshrma 
Our target was achieved with only one SSM association and an event bridge rule, therefore removed the other association.

1. SSM Association - This association is applied automatically when CFT is deployed for the first time in an AWS account, thereby covering all the existing EC2 instances. It also updates the instances with latest version NewRelic infrastructure agent once every month.
2. Event Bridge Rule - Whenever an instance goes in running state, this rule gets triggered and again latest version NR agent is installed in all instances. The destination of the rule is AWS-ConfigureAWSPackage. 

Note:
Triggering SSM Association on EC2 state change is not possible using event bridge rule.

Please let me know if I'm missing something.